sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -120,6 +120,22 @@ def check_max_n_features(est, X, y):
         assert_equal(X_new.shape[1], n_features)
 
 
+def check_threshold_and_max_n_feature(est, X, y):
+    transformer1 = SelectFromModel(estimator=est, max_n_features=3)
+    transformer1.fit(X, y)
+    X_new1 = transformer1.transform(X)
+
+    transformer2 = SelectFromModel(estimator=est, threshold=0.1)
+    transformer2.fit(X, y)
+    X_new2 = transformer2.transform(X)
+
+    transformer3 = SelectFromModel(estimator=est, max_n_features=3,
+                                   threshold=0.1)
+    transformer3.fit(X, y)
+    X_new3 = transformer3.transform(X)
+    assert_equal(X_new3.shape[1], min(X_new1.shape[1], X_new2.shape[1]))
```

Comment:
`np.arange(X.shape[1])[np.newaxis, :]` is just producing `[[0, 1, 2, ... n_features]]`.